### PR TITLE
#3513: Align icons in Blueprints sidebar

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Reduces wrapping by using space better
- Avoids wrapping of text _below_ the icon
- Part of https://github.com/pixiebrix/pixiebrix-extension/issues/3513

## Before / After

Screen Shot 36 Screen Shot 35

## Checklist

- [x] https://github.com/pixiebrix/pixiebrix-extension/issues/4456
- [x] Designate a primary reviewer: @mnholtz

### CODE REVIEW

File: src/options/pages/blueprints/ListFilters.module.scss [hunk 1]
```diff
@@ -6,4 +6,8 @@
     font-size: 0.8rem;
     color: #bba8bff5;
   }
+
+  :global .nav-link {
+    padding-inline: 0.7rem; // Reduce horizontal padding
```
Comment (fregante) [inline src/options/pages/blueprints/ListFilters.module.scss:11]: Reduces the padding inside the selected elements, which at 1rem seems excessive, compared to the available space

File: src/options/pages/blueprints/ListFilters.tsx [hunk 2]
```diff
@@ -66,6 +70,16 @@ export const BLUEPRINTS_PAGE_TABS: BlueprintTabMap = {
...
 };
 
+const ListItem: BsPrefixRefForwardingComponent<
+  "a",
+  NavLinkProps & { label: string; icon: IconProp }
+> = ({ label, icon, ...otherProps }) => (
+  <Nav.Link className="media" {...otherProps}>
+    <FontAwesomeIcon className="align-self-center" icon={icon} />
+    <span className="media-body ml-2">{label}</span>
+  </Nav.Link>
```
Comment (fregante) [inline src/options/pages/blueprints/ListFilters.tsx:80]: We should use this pattern in more places:

https://getbootstrap.com/docs/4.0/layout/media-object/

Potentially `ListItem` could be extracted and reused elsewhere.

Note: `Nav.Item` was unnecessary and was dropped.

File: src/options/pages/blueprints/ListFilters.tsx [hunk 3]
```diff
@@ -66,6 +70,16 @@ export const BLUEPRINTS_PAGE_TABS: BlueprintTabMap = {
   },
 };
 
+const ListItem: BsPrefixRefForwardingComponent<
+  "a",
+  NavLinkProps & { label: string; icon: IconProp }
+> = ({ label, icon, ...otherProps }) => (
```
Comment (fregante) [inline src/options/pages/blueprints/ListFilters.tsx:76]: I couldn't find a better way to type this. 

- `NavLink` isn't generic and can't be extended
- `React.FC<NavLinkProps>` for some reason lacks link attributes like `href`
- `React.FC<NavLink>` lacks `eventKey`, which is part of NavLinkProps
- `React.FC<NavLink & NavLinkProps>` is the same as above

File: src/options/pages/blueprints/ListFilters.tsx [hunk 4]
```diff
@@ -157,7 +171,7 @@ const ListFilters: React.FunctionComponent<ListFiltersProps> = ({
 
   return (
     <Col sm={12} md={3} xl={2} className={styles.root}>
-      <Form className="mr-3">
```
Comment (fregante) [inline src/options/pages/blueprints/ListFilters.tsx:160]: This margin was unnecessarily reducing the available horizontal space. Bootstrap’s columns already add a lot of space. Was it added to solve any specific issues?

File: src/options/pages/blueprints/ListFilters.tsx (hunk 2 above)
Comment (mnholtz) [reply src/options/pages/blueprints/ListFilters.tsx:80]: Good point! Never thought to apply this pattern to vertical menus before.

File: src/options/pages/blueprints/ListFilters.tsx (hunk 4 above)
Comment (mnholtz) [reply src/options/pages/blueprints/ListFilters.tsx:160]: Not sure. It was probably me at some point. Could be an artifact of a spacing issue that got resolved. But I don't see any reason to keep it anymore.

**Review by mnholtz — APPROVED** [review]
I love this!!

**Pull request merged**